checkov/secrets/plugins/custom_regex_detector.py

```diff
@@ -43,26 +50,85 @@ def analyze_line(
+                **kwargs
+            )
+
+        return output
+
+    def _find_potential_secret(
+            self,
+            filename: str,
+            string_to_analyze: str,
+            output: Set[PotentialSecret],
+            line_number: int = 0,
+            context: Optional[CodeSnippet] = None,
+            is_multiline: bool = False,
+            **kwargs: Any
+    ) -> None:
+        current_denylist: Set[Pattern[str]] = self.multiline_deny_list if is_multiline else self.denylist
+        current_regex_to_metadata: dict[str, dict[str, Any]] = self.multiline_regex_to_metadata if is_multiline else self.regex_to_metadata
+        kwargs["regex_denylist"] = current_denylist
+        for match, regex in self.analyze_string(string_to_analyze, **kwargs):
             try:
                 verified_result = call_function_with_arguments(self.verify, secret=match, context=context)
                 is_verified = True if verified_result == VerifiedResult.VERIFIED_TRUE else False
             except Exception:
                 is_verified = False
-            regex_data = self.regex_to_metadata[regex.pattern]
-            ps = PotentialSecret(type=regex_data["Name"], filename=filename, secret=match,
-                                 line_number=line_number, is_verified=is_verified)
-            ps.check_id = self.regex_to_metadata[regex.pattern]["Check_ID"]  # type:ignore[attr-defined]
-            if len(cast(str, ps.secret_value)) in range(MIN_CHARACTERS, MAX_CHARACTERS) or not regex_data['isCustom']:
+            regex_data = current_regex_to_metadata[regex.pattern]
+            # Wrap multiline match with fstring + ''
+            match = f"'{match}'" if is_multiline else match
+            ps = PotentialSecret(
+                type=regex_data["Name"],
+                filename=filename,
+                secret=match,
+                line_number=line_number,
+                is_verified=is_verified
+            )
+            ps.check_id = current_regex_to_metadata[regex.pattern]["Check_ID"]  # type:ignore[attr-defined]
+            if is_multiline:
+                output.add(ps)
+            elif len(cast(str, ps.secret_value)) in range(MIN_CHARACTERS, MAX_CHARACTERS) or not regex_data['isCustom']:
                 output.add(ps)
             else:
                 logging.info(
                     f'Finding for check {ps.check_id} are not 5-100 characters in length, was ignored')  # type: ignore
 
-        return output
-
-    def analyze_string(self, string: str, **kwargs: Optional[Dict[str, Any]]) -> Generator[Tuple[str, Pattern[str]], None, None]:  # type:ignore[override]
-        for regex in self.denylist:
+    def analyze_string(self, string: str, **kwargs: Optional[Dict[str, Any]]) -> Generator[Tuple[str, Pattern[str]], None, None]:  # type: ignore # type:ignore[override]
+        regex_denylist: Optional[Set[Pattern[str]]] = set()
+        regex_denylist = kwargs.get("regex_denylist", self.denylist)  # type: ignore
```

Comment:
Let's try to reduce mypy ignores please 🙏 . Also, please don't do `# type: ignore` as it is a general, dangerous suppression
```suggestion
    def analyze_string(self, string: str, **kwargs: Optional[Dict[str, Any]]) -> Generator[Tuple[str, Pattern[str]], None, None]:  # type:ignore[override]
        regex_denylist: Set[Pattern[str]] = kwargs.get("regex_denylist") or self.denylist  # type:ignore[assignment]
        for regex in regex_denylist:
```

Reply:
talked about it   :)

Reply:
Still need to take the mypy reductions @pazbechor